### PR TITLE
Fix types

### PR DESCRIPTION
**Pull request by fzimmermann89**
This fixes all type issues in tests and some other inconsistencies.
- Make the KData mixins return a Self (i.e. KData). So, for example, KData.remove_oversampling will now result in KData (used to be Any)
- Fix issues caused by using numpy in the rotation tests. Widen allowed types in Rotation from only tensor inputs to also allow nested sequences of floats.
- Add type ignores for paths that can never be reached according to the type hints but are tested anyways
- Allow a single int as size for RandomGenerator
- Make return_sensitivity in Rotation.align_vectors a keyword-only argument, and overload the return type conditional on its value

DO NOT CLICK THE MERGE BUTTON without considering the order of merges according to the stack..

### CODE REVIEW

**Comment by fzimmermann89** [comment]
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #283
* __->__ #282
* #280
* #279

File: src/mrpro/phantoms/_EllipsePhantom.py [hunk 1]
```diff
@@ -91,7 +91,7 @@ def image_space(self, image_dimensions: SpatialDimension[int]) -> torch.Tensor:
         """
         # Calculate image representation of phantom
         ny, nx = image_dimensions.y, image_dimensions.x
-        ix, iy = torch.meshgrid(
+        ix, it = torch.meshgrid(
```
Comment (ckolbPTB) [inline src/mrpro/phantoms/_EllipsePhantom.py]: why `it`? also in line 104 below

**Review by ckolbPTB — CHANGES_REQUESTED** [review]

File: src/mrpro/phantoms/_EllipsePhantom.py (hunk 1 above)
Comment (fzimmermann89) [reply src/mrpro/phantoms/_EllipsePhantom.py]: spelling autocorrect fail of the precommit .... iy->it
we should add this as an exception....

Comment (fzimmermann89) [reply src/mrpro/phantoms/_EllipsePhantom.py]: should be fixed now

**Review by ckolbPTB — APPROVED** [review]

**Pull request merged**